lib/semantics/resolve-names.cc

```diff
@@ -2245,20 +2264,27 @@ void DeclarationVisitor::Post(const parser::ProcInterface &x) {
 }
 
 void DeclarationVisitor::Post(const parser::ProcDecl &x) {
-  const auto &name{std::get<parser::Name>(x.t).source};
+  bool isFunction{false};
+  bool isSubroutine{false};
   ProcInterface interface;
   if (interfaceName_) {
     if (auto *symbol{FindExplicitInterface(*interfaceName_)}) {
       interface.set_symbol(*symbol);
+      isFunction = symbol->test(Symbol::Flag::Function);
+      isSubroutine = symbol->test(Symbol::Flag::Subroutine);
     }
   } else if (auto &type{GetDeclTypeSpec()}) {
     interface.set_type(*type);
+    isFunction = true;
   }
   auto attrs{GetAttrs()};
   if (currScope().kind() != Scope::Kind::DerivedType) {
     attrs.set(Attr::EXTERNAL);
   }
-  DeclareProcEntity(name, attrs, interface);
+  const auto &name{std::get<parser::Name>(x.t).source};
```

Comment:
I'd feel more comfortable if there was an assertion indicating that only one of isF or isS can get true, if that's true.

Reply:
I'll look into adding consistency checking for the flags when they are set. There are probably other things that could be checked.

Reply:
Few of the current set of `Symbol` flags can be simultaneously set and maybe it could become an `enum`.

Reply:
It turns out that setting Function or Subroutine on the symbol here is not necessary becuase DeclareProcEntity takes care of it.